### PR TITLE
Replace mailhog with mailpit

### LINKED ISSUE #257: Switch from mailhog to mailpit
Mailhog is not really maintained anymore. We should switch to something else.
Mailpit is: https://github.com/axllent/mailpit
Plus it has multiarch docker images, and we're experimenting using it internally to help automate mail sending as part of behat.

### PR DESCRIPTION
**Pull request by mattporritt**
mailpit (https://github.com/axllent/mailpit) is a more maintained mail test client with a straightforward api.

Fixes #257

### CODE REVIEW

**Comment by mattporritt** [comment]
This replaces mailhog with mailpit. It is more or less a drop in replacement.

File: README.md [hunk 1]
```diff
@@ -128,7 +128,7 @@ bin/moodle-docker-compose exec webserver php admin/cli/install_database.php --ag
 
 Notes:
 * Moodle is configured to listen on `http://localhost:8000/`.
-* Mailhog is listening on `http://localhost:8000/_/mail` to view emails which Moodle has sent out.
+* Mailpit is listening on `http://localhost:8025/_/mail` to view emails which Moodle has sent out.
```
Comment (stronk7) [inline README.md:131]:
```suggestion
* Mailpit is listening on `http://localhost:8000/_/mail` to view emails which Moodle has sent out.
```

I think `8000` is the correct public address that apache is proxying from internal (not published) `8025`.

**Review by stronk7 — COMMENTED** [review]
Hi,

I think that with the tiny change suggested, this is pretty acceptable.

Nice one, thanks!

File: README.md (hunk 1 above)
Comment (mattporritt) [reply README.md]: Ahh, thanks for the pickup. That is a leftover from an earlier version of the patch. I have fixed it now and force pushed

Cheers,
Matt P

**Review by andrewnicols — APPROVED** [review]

**Comment by skodak** [comment]
this is a very annoying change, please do not change the contents of config.php unnecessarily, and if you do then please use more future-proof setting and docker image names such as:

`$CFG->smtphosts = 'mail:1025';`

**Pull request merged**